src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,18 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    private boolean isPrivate;
+
+	public Contact(boolean isPrivate) {
+		super();
```

Comment:
does it make sense for the Contact class to not contain a value like a name? As mentioned in the tutorial discussions, the entire constructor code from the child classes can be pushed up to this parent class, including the validation methods and the validation regex